File: src/equiv/ClimbingStairs3.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "ClimbingStairs3"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "climbStairs2"
            "climbStairs3"
            [x]
            [E.mk_ige x (E.Int 2)]

File: benchmark/equiv/ClimbingStairs3.java
public class ClimbingStairs3 {
  static public int climbStairs2(int n) {
    int sum = 2, prev = 1, curr = 0;
    for (int i = 2; i < n; i++) {
      curr = sum;
      sum += prev;
      prev = curr;
    }
    return sum;
  }
  static public int climbStairs3(int n) {
    int p1 = 1, p2=1, curr=1;

    for(int i=2;i<=n;i++){
      curr = p1 + p2;
      p2 = p1;
      p1 = curr;
    }
    return curr;
  }
}
